airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -1138,7 +1139,7 @@ def execute(self, context: Context):
                 impersonation_chain=self.impersonation_chain,
             )
         if isinstance(self.sql, str):
-            job_id: str | list[str] = self.hook.run_query(
+            self.job_id = self.hook.run_query(
```

Comment:
Yes. I think it's better to store it as attribute - this means no unnecessary call to xcom backend, as we're inside the same process when calling the openlineage method.

Reply:
There can be a problem to this. `self.job_id` is populated only in execute() method. This might not be available in get_openlineage_facet_on_start(). WDYT?

Reply:
@sunank200 that's true. That's why only `get_openlineage_facets_on_complete` method is implemented - it's not possible to get this data from `BigQuery` API prior to job starting.